### PR TITLE
[AppService]: Feature #19496 Added -Tag parameter support to New-AzWebApp & New-AzWebAppSlot  

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
…pSlot

#19496 
## Description
In my organisation we are required to add tags for every Azure resource. The mentioned cmdlet doesn't have a -Tag parameter which effectively prevents me from creating a resource. Ont he other hand Azure Portal also doesn't allow me to supply the tag upon creation.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Websites/Websites.Test/ScenarioTests/WebAppTests.ps1 [hunk 1]
```diff
@@ -422,20 +422,23 @@ function Test-CreateNewWebApp
...
-		$job = New-AzWebApp -ResourceGroupName $rgname -Name $wname -Location $location -AppServicePlan $whpName -AsJob
+		$job = New-AzWebApp -ResourceGroupName $rgname -Name $wname -Location $location -AppServicePlan $whpName -Tag $tag -AsJob
 		$job | Wait-Job
 		$actual = $job | Receive-Job
 		
 		# Assert
 		Assert-AreEqual $wname $actual.Name
 		Assert-AreEqual $serverFarm.Id $actual.ServerFarmId
+		Assert-AreEqual $tag.Keys $actual.Tags.Keys
+        Assert-AreEqual $tag.Values $actual.Tags.Values
```
Comment (Nickcandy) [inline src/Websites/Websites.Test/ScenarioTests/WebAppTests.ps1:441]: Please fix the indents

**Comment by Nickcandy** [comment]
It looks good to me except the indentation. Please help me double-check @BethanyZhou

File: src/Websites/Websites.Test/ScenarioTests/WebAppTests.ps1 (hunk 1 above)
Comment (Kotasudhakarreddy) [reply src/Websites/Websites.Test/ScenarioTests/WebAppTests.ps1]: Done!

File: src/Websites/Websites/ChangeLog.md [hunk 2]
```diff
@@ -18,7 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
-
+* Added Tag parameter for `New-AzWebApp` and `New-AzWebAppSlot`
```
Comment (BethanyZhou) [inline src/Websites/Websites/ChangeLog.md:21]:
```suggestion
* Added Tag parameter for `New-AzWebApp` and `New-AzWebAppSlot`

```
Leave a space before next section

File: src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs [hunk 3]
```diff
@@ -100,6 +100,9 @@ public class NewAzureWebAppSlotCmdlet : WebAppBaseClientCmdLet
 		[Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
```
Comment (BethanyZhou) [inline src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs:101]:
```suggestion
		public SwitchParameter AsJob { get; set; }
```
Correct indent please

File: src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs [hunk 4]
```diff
@@ -181,7 +184,7 @@ public override void ExecuteCmdlet()
             }
 
             var webApp = new PSSite(WebsitesClient.GetWebApp(ResourceGroupName, Name, null));
-			var site = new PSSite(WebsitesClient.CreateWebApp(ResourceGroupName, Name, Slot, webApp.Location, AppServicePlan==null?webApp.ServerFarmId : AppServicePlan, cloningInfo, AseName, AseResourceGroupName));
+			var site = new PSSite(WebsitesClient.CreateWebApp(ResourceGroupName, Name, Slot, webApp.Location, AppServicePlan==null?webApp.ServerFarmId : AppServicePlan, cloningInfo, AseName, AseResourceGroupName, (IDictionary<string, string>)CmdletHelpers.ConvertToStringDictionary(Tag)));
 			UpdateConfigIfNeeded(site);
```
Comment (BethanyZhou) [inline src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs:188]: Please keep the L186 and L187-188 with same indents.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

**Review by BethanyZhou — APPROVED** [review]
Looks good to me

**Comment by BethanyZhou** [comment]
Hi @Poolshark314 , could you append -Debug at the end of your command and paste the debug message here?

**Comment by Poolshark314** [comment]
@BethanyZhou Unfortunately the output looks the same when run with -Debug:

New-AzWebApp : Parameter set cannot be resolved using the specified named parameters.
At line:1 char:11
+ $webApp = New-AzWebApp -Name $sitename -ResourceGroupName $RGName -Lo ...
+           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [New-AzWebApp], ParameterBindingException
    + FullyQualifiedErrorId : AmbiguousParameterSet,Microsoft.Azure.Commands.WebApps.Cmdlets.WebApps.NewAzureWebAppCmdlet

**Comment by BethanyZhou** [comment]
Hi @Poolshark314 , checked the syntax of New-AzWebApp, I find -Tag only exists in the first parameter set.

See 
```powershell
New-AzWebApp [[-ResourceGroupName] <String>] [-Name] <String> [[-Location] <String>]
 [[-AppServicePlan] <String>] [-ContainerImageName <String>] [-EnableContainerContinuousDeployment] [-AsJob]
 [-GitRepositoryPath <String>] [-Tag <Hashtable>] [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
 [-Confirm] [<CommonParameters>]

New-AzWebApp [-ResourceGroupName] <String> [-Name] <String> [[-Location] <String>] [[-AppServicePlan] <String>]
 -ContainerImageName <String> -ContainerRegistryUrl <String> -ContainerRegistryUser <String>
 -ContainerRegistryPassword <SecureString> [-EnableContainerContinuousDeployment] [-AsJob]
 [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]

New-AzWebApp [-ResourceGroupName] <String> [-Name] <String> [-Location] <String> [[-AppServicePlan] <String>]
 [[-SourceWebApp] <PSSite>] [[-TrafficManagerProfile] <String>] [-EnableContainerContinuousDeployment]
 [-IgnoreSourceControl] [-IgnoreCustomHostNames] [[-AppSettingsOverrides] <Hashtable>] [[-AseName] <String>]
 [[-AseResourceGroupName] <String>] [-IncludeSourceWebAppSlots] [-AsJob]
 [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
Which means AseResourceGroupName and AseName (in last parameter set) can't work with Tag together in New-AzWebApp. @Kotasudhakarreddy  could you check is this behavior by design?

**Comment by Kotasudhakarreddy** [comment]
@BethanyZhou - Yes, it is by design. 
@Poolshark314 please raise a new bug to extend this feature for all the `ParameterSetNames`.

**Pull request merged**